### PR TITLE
Block Directory: Remove 'edit-post' package dependency

### PR DESCRIPTION
**Pull request by Mamaduka**
## What?
This PR removes `@wordpress/edit-post` as a dependency from the block directory package. Component usage from `edit-post` was discouraged after #45895, and it was probably a leftover from the "great unification."

I've also fixed the import for the `PluginPrePublishPanel` slot. The editor package is already used here.

## Testing Instructions
The feature has e2e test coverage.

1. Create a post.
2. Install a block plugin from the directory.
3. Try publishing a post.
4. Confirm that the newly installed block is listed in the pre-publish sidebar.

### Testing Instructions for Keyboard
Same.

## Screenshots or screencast 
CleanShot 2024-07-10 at 13 29 24

### CODE REVIEW

File: packages/block-directory/src/plugins/installed-blocks-pre-publish-panel/index.js [hunk 1]
```diff
@@ -4,17 +4,14 @@
 import { _n, sprintf } from '@wordpress/i18n';
 import { useSelect } from '@wordpress/data';
 import { blockDefault } from '@wordpress/icons';
+import { PluginPrePublishPanel } from '@wordpress/editor';
```
Comment (Mamaduka) [inline packages/block-directory/src/plugins/installed-blocks-pre-publish-panel/index.js:7]: Ref - https://github.com/WordPress/gutenberg/pull/60628#discussion_r1559561649

File: packages/block-directory/src/plugins/installed-blocks-pre-publish-panel/index.js [hunk 2]
```diff
@@ -4,17 +4,14 @@
...
 
 /**
  * Internal dependencies
  */
 import CompactList from '../../components/compact-list';
 import { store as blockDirectoryStore } from '../../store';
 
-// We shouldn't import the editor package directly
-// because it would include the wp-editor in all pages loading the block-directory script.
```
Comment (ellatrix) [inline packages/block-directory/src/plugins/installed-blocks-pre-publish-panel/index.js:15]: What other pages is this referring to?

**Review by ellatrix — APPROVED** [review]

File: packages/block-directory/src/plugins/installed-blocks-pre-publish-panel/index.js (hunk 2 above)
Comment (Mamaduka) [reply packages/block-directory/src/plugins/installed-blocks-pre-publish-panel/index.js:15]: The original comment referred to the `wp-edit-post` package, accidentally loading it on other pages like Site Editor.

The context got lost somewhere in a string replacing. Full details can be found here - #45895.

**Comment by Mamaduka** [comment]
Thanks, Ella!

**Pull request merged**